### PR TITLE
controller wrappers

### PR DESCRIPTION
**Pull request by ri-nath**
motor controller wrappers with configuration in constructor & lazy set methods

### CODE REVIEW

File: src/main/java/org/team4159/lib/hardware/controller/rev/CardinalSpark.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.ControlType;
+
+public class CardinalSpark extends CANSparkMax {
+  public static final int DEFAULT_CURRENT_LIMIT = 40;
+
+  private double last_setpoint;
+  private ControlType last_control_type;
+
+  public CardinalSpark(int id, IdleMode mode) {
```
Comment (kaijchang) [inline src/main/java/org/team4159/lib/hardware/controller/rev/CardinalSpark.java]: can we add support for Brushed mode too just in case

**Review by kaijchang — CHANGES_REQUESTED** [review]
op op

File: src/main/java/org/team4159/frc/robot/subsystems/Drivetrain.java [hunk 2]
```diff
@@ -1,6 +1,7 @@
 package org.team4159.frc.robot.subsystems;
 
 import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj.Talon;
```
Comment (kaijchang) [inline src/main/java/org/team4159/frc/robot/subsystems/Drivetrain.java]: unused import

File: src/main/java/org/team4159/lib/hardware/controller/rev/CardinalMAX.java [hunk 3]
```diff
@@ -3,18 +3,22 @@
...
 
-  public CardinalSpark(int id, IdleMode mode) {
-    this(id, mode, DEFAULT_CURRENT_LIMIT);
+  public CardinalMAX(int id, IdleMode mode) {
+    this(id, mode, DEFAULT_CURRENT_LIMIT, MotorType.kBrushless);
   }
 
-  public CardinalSpark(int id, IdleMode mode, int current_limit) {
-    super(id, MotorType.kBrushless);
+  public CardinalMAX(int id, IdleMode mode, int current_limit) {
```
Comment (kaijchang) [inline src/main/java/org/team4159/lib/hardware/controller/rev/CardinalMAX.java]: this doesn't use `current_limit`, but i think you can probably remove it anyways so it's just the general constructor and then the one we commonly use

File: src/main/java/org/team4159/frc/robot/subsystems/Drivetrain.java (hunk 2 above)
Comment (ri-nath) [reply src/main/java/org/team4159/frc/robot/subsystems/Drivetrain.java]: o

**Review by kaijchang — CHANGES_REQUESTED** [review]

**Review by kaijchang — APPROVED** [review]

**Pull request merged**